File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        
        while(!s.equals("1")) {
            answer[0]++;
            int zeroCount = 0;
            
            StringBuilder x = new StringBuilder();
            for(char c : s.toCharArray()) {
                if(c == '1') {
                    x.append("1");
                }
                else {
                    zeroCount++;
                }
            }
            
            answer[1] += zeroCount;
            s = Integer.toBinaryString(x.length());
        }
        
        return answer;
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    answer.append(0)
    answer.append(0)
    
    while s != '1':
        answer[0] += 1
        answer[1] += s.count('0')
        
        x = ''
        for i in s:
            if i == '1':
                x += i
        
        s = format(len(x), 'b')
        
    return answer